packages/@coorpacademy-components/src/molecule/base-modal/index.js

```diff
@@ -0,0 +1,132 @@
+import React, {useCallback} from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../../atom/icon';
+import ButtonLink from '../../atom/button-link';
+import style from './style.css';
+
+const BaseModal = props => {
+  const {title, description, headerIcon, children, isOpen, footer, onClose} = props;
+
+  const Footer = useCallback(() => {
+    if (!footer) return null;
+    if (typeof footer === 'function') return footer();
+
+    const {cancelButton, confirmButton, text, isError} = footer;
+
+    if (!cancelButton && !confirmButton && !text) return null;
+
+    return (
+      <div className={style.footer}>
+        <div className={style.footerCTAWrapper}>
+          {cancelButton?.onCancel && cancelButton?.label ? (
```

Comment:
you can destructure above to avoid repetition?
```
const {onCancel, label: cancelLabel} = cancelButton;
```
```suggestion
          {onCancel && cancelLabel ? (
```